File: isegoria/src/main/java/com/eulersbridge/isegoria/models/CandidateTicket.java
package com.eulersbridge.isegoria.models;

import android.text.TextUtils;

import com.google.gson.annotations.SerializedName;

/**
 * Created by Seb on 04/11/2017.
 */

public class CandidateTicket {

    @SerializedName("ticketId")
    public long id;

    private String name;
    private String givenName;
    private String familyName;

    @SerializedName("numberOfSupporters")
    public String supportersCount;

    public String information;
    public String logo;

    private String colour;

    public String getName() {
        if (!TextUtils.isEmpty(name)) {
            return name;

        } else {
            return String.format("%s %s", givenName, familyName);
        }
    }

    public String getColour() {
        if (!TextUtils.isEmpty(colour)) {
            return colour;

        } else {
            return "#000000";
        }
    }
}

File: isegoria/src/main/kotlin/com/eulersbridge/isegoria/network/api/model/CandidateTicket.kt
package com.eulersbridge.isegoria.network.api.model

import com.squareup.moshi.Json

data class CandidateTicket (

    @field:Json(name = "ticketId")
    var id: Long = 0,

    var name: String?,
    var givenName: String?,
    var familyName: String?,

    @field:Json(name = "numberOfSupporters")
    var supportersCount: String?,

    var information: String?,
    var logo: String? = null,
    var code: String?,

    private val colour: String?

) {
    val fullName: String
        get() = name?.takeUnless { it.isBlank() } ?: "$givenName $familyName"
    
    fun getColour()
        = colour?.takeUnless { it.isBlank() } ?: "#000000"
}